opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -52,37 +56,54 @@
    *
    * @param <C> the carrier type, which also parametrizes the Format.
    * @param spanContext the SpanContext instance to inject into the carrier
-   * @param format the Format of the carrier
-   * @param carrier the carrier for the SpanContext state. All Tracer.inject() implementations must support io.opentracing.propagation.TextMap and java.nio.ByteBuffer.
+   * @param format the Format of the carrier. All Tracer.inject() implementations
+   *               must support formats defined in io.opentracing.propagation.Format.Builtin
+   * @param carrier the carrier for the SpanContext state.
+   *
+   * @throws io.opentracing.propagation.UnsupportedFormatException when tracer does not support given format
    *
    * @see io.opentracing.propagation.Format
    * @see io.opentracing.propagation.Format.Builtin
+   * @see io.opentracing.propagation.UnsupportedFormatException
    */
   <C> void inject(SpanContext spanContext, Format<C> format, C carrier);
 
   /**
    * Extract a SpanContext from a `carrier` of a given type, presumably after propagation across a process boundary.
    *
+   * If tracer does not support given format, it must throw
+   * io.opentracing.propagation.UnsupportedFormatException so that instrumentation
+   * can try another format.
+   *
+   * If given carrier does not contain information to construct SpanContext,
+   * the tracer must return `null`. If SpanContext cannot be extracted because
+   * the data in the carrier is malformed, the tracer should return null, rather
+   * than throw an exception. The tracer may handle the exceptions internally
+   * by logging them or incrementing an error counter.
+   *
    * <p>Example:
    * <pre>{@code
    * Tracer tracer = ...
    * TextMap httpHeadersCarrier = new AnHttpHeaderCarrier(httpRequest);
    * SpanContext spanCtx = tracer.extract(Format.Builtin.HTTP_HEADERS, httpHeadersCarrier);
-   * tracer.buildSpan('...').withChildOf(spanCtx).start();
+   * tracer.buildSpan('...').asChildOf(spanCtx).start();
    * }</pre>
    *
    * If the span serialized state is invalid (corrupt, wrong version, etc) inside the carrier this will result in an
    * IllegalArgumentException.
    *
    * @param <C> the carrier type, which also parametrizes the Format.
-   * @param format the Format of the carrier
-   * @param carrier the carrier for the SpanContext state. All Tracer.extract() implementations must support
-   *                io.opentracing.propagation.TextMap and java.nio.ByteBuffer.
+   * @param format the Format of the carrier. All Tracer.extract() implementations
+   *               must support formats defined in io.opentracing.propagation.Format.Builtin
+   * @param carrier the carrier for the SpanContext state.
    *
-   * @return the SpanContext instance holding context to create a Span.
+   * @return the SpanContext instance extracted from the carrier holding context to create a Span or null if it cannot be extracted.
```

Comment:
I'd be in favor of adding something like this:

```java
public interface SpanContext {
  SpanContext NONE = NullSpanContext.INSTANCE;
  
   Iterable<Map.Entry<String, String>> baggageItems();
}

/*package-private*/ enum NullSpanContext implements SpanContext {
  INSTANCE;

  @Override public Iterable<Map.Entry<String, String>> baggageItems() {
    return Collections.emptySet();
  }
}
```

With this API, you can specify that implementations disregard any attempt to add a reference that uses `SpanContext.NONE` is no-op; the following code works without change whether a context was present or not:

```java
SpanContext spanCtx = tracer.extract(Format.Builtin.HTTP_HEADERS, httpHeadersCarrier);
tracer.buildSpan('...').asChildOf(spanCtx).start();
```

API users that do want to know if the incoming context actually had a tracing context or not can still check for it by reference equality:

```java
SpanContext spanCtx = tracer.extract(Format.Builtin.HTTP_HEADERS, httpHeadersCarrier);
if (spanCtx == SpanContext.NONE) {
 // handle missing context
}
```

The `enum` construct is there to prevent multiple instances of a null context from being created in some edge cases.